### DISCUSSION TITLE
Symmetry in indices of sparse data

### DISCUSSION THREAD

**Discussion by scemama** [TREX-CoE/trexio#154] [Ideas]
We could simply give the list of equivalent indices obtained by permutations. This would imply the automatic creation of a variable `nperm` containing the number of equivalent permutations and `perm`, the list of permutations.
The default is `nperm=1` and `perm=[(1,2,3,4)]` (for 4 index data). We can encode the potential sign change in the sign of the first element.
 
For example, for two-e integrals:
```
int32_t perm[8][4] = {
  1, 2, 3, 4, 
  3, 2, 1, 4, 
  1, 4, 3, 2, 
  3, 4, 1, 2,
  2, 1, 4, 3,                                                                                                                
  2, 3, 4, 1, 
  4, 1, 2, 3, 
  4, 3, 2, 1 };

trexio_write_ao_2e_int_eri_perm(trexio_file, 8, perm);
```

We could also store the most common predefined values:
```
int32_t trexio_4_index_1_perm[1][4] = {
  1, 2, 3, 4 };

int32_t trexio_6_index_1_perm[1][6] = {
  1, 2, 3, 4, 5, 6 };

int32_t trexio_4_index_4_perm_real[4][4] = {
  1, 2, 3, 4, 
  3, 2, 1, 4, 
  1, 4, 3, 2, 
  3, 4, 1, 2 };

int32_t trexio_4_index_4_perm_cplx[4][4] = {
  1, 2, 3, 4, 
  -3, 2, 1, 4, 
  -1, 4, 3, 2, 
  3, 4, 1, 2 };

int32_t trexio_4_index_8_perm_real[8][4] = {
  1, 2, 3, 4, 
  3, 2, 1, 4, 
  1, 4, 3, 2, 
  3, 4, 1, 2,
  2, 1, 4, 3,                                                                                                                
  2, 3, 4, 1, 
  4, 1, 2, 3, 
  4, 3, 2, 1 };

int32_t trexio_4_index_8_perm_cplx[8][4] = {
  1, 2, 3, 4, 
  -3, 2, 1, 4, 
  -1, 4, 3, 2, 
  3, 4, 1, 2,
  2, 1, 4, 3,                                                                                                                
  -2, 3, 4, 1, 
  -4, 1, 2, 3, 
  4, 3, 2, 1 };

```

**Comment by q-posev** [comment]
Looks good to me but I would replace `nperm` with `perm_num` in order to be consistent with other dimensioning variables. Perhaps @plopezrios or @michalhapka could comment on this from the user point of view.

**Comment by michalhapka** [comment]
Dear all,
I think it is an elegant and convenient way of passing the symmetry information, the predefined values also look useful.
